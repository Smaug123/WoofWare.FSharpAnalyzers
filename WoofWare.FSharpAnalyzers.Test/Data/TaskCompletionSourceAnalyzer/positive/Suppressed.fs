module TaskCompletionSourceSuppressed

open System.Threading.Tasks

let createTcsSuppressed () =
    // ANALYZER: RunContinuationsAsynchronously explicitly skipped here
    let tcs = TaskCompletionSource<int> ()
    tcs.SetResult 42
    tcs.Task

let createTcsNotSuppressed () =
    let tcs = TaskCompletionSource<int> ()
    tcs.SetResult 99
    tcs.Task
