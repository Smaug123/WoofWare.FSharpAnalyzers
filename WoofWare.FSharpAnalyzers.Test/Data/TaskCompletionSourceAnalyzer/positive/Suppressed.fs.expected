WOOF-TCS-ASYNC | Warning | (12,14--12,42) | TaskCompletionSource<T> created without TaskCreationOptions.RunContinuationsAsynchronously. This can cause continuations to run inline on the calling thread, leading to deadlocks, thread-pool starvation, and corruption of state. Always use: new TaskCompletionSource<T>(TaskCreationOptions.RunContinuationsAsynchronously). Suppress with comment including text 'TaskCompletionSource without RunContinuationsAsynchronously allowed'.
